### PR TITLE
Add support for presets

### PR DESCRIPTION
**Pull request by wooorm**
* Update for remark-lint@5.0.0;
* Atom setting: whether to use ignore files;
* Atom setting: whether to use remark-preset-lint-recommended
  if `remark-lint` is not configured;
* Atom setting: whether to use remark-preset-lint-consistent
  if `remark-lint` is not configured;
* Atom setting: customise scopes to process;
* Add more docs;
* Update tests, dependencies.

/cc @Arcanemagus

---
This change is [Reviewable](https://reviewable.io/reviews/atomlinter/linter-markdown/106)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -9,10 +9,21 @@
 [![Plugin installs!](https://img.shields.io/apm/dm/linter-markdown.svg)](https://atom.io/packages/linter-markdown)
 [![Package version!](https://img.shields.io/apm/v/linter-markdown.svg?style=flat)](https://atom.io/packages/linter-markdown)
 
-Lint markdown files utilizing [remark-lint][remark-lint] and the
+Lint markdown files using [remark-lint][remark-lint] and the
 [linter][linter] package for atom.
 
-The linting is easily [configurable][configuration] with remark-lint.
+If there is no configuration for **remark-lint**, this linter runs [remark-preset-lint-consistent][consistent] and
```
Comment (Arcanemagus) [inline README.md]: "no configuration **found** for"

File: lib/index.js [hunk 2]
```diff
@@ -8,30 +8,40 @@
...
+let engine;
+let processor;
 
 /**
  * Linter-markdown.
  *
  * @return {LinterConfiguration}
  */
 function linter() {
+  const config = atom.config.get('linter-markdown');
```
Comment (Arcanemagus) [inline lib/index.js]: Why not observe `linter-markdown.detectIgnore` in the `activate()` method? There isn't a reason to do all the parsing required to grab the entire config here.

File: lib/index.js [hunk 3]
```diff
@@ -48,6 +58,43 @@ function activate() {
  */
 module.exports = {
   activate,
-  config: {},
-  provideLinter: linter
+  provideLinter: linter,
+  config: {
```
Comment (Arcanemagus) [inline lib/index.js]: Use a `configSchema` in `package.json` ([example](https://github.com/AtomLinter/linter-stylelint/blob/v3.1.0/package.json#L33)) so Atom can know about the configuration without having to run the package code.

**Comment by Arcanemagus** [comment]
That's all I see for now 😉.

**Comment by wooorm** [comment]
Done!

**Comment by Arcanemagus** [comment]
Looks like CI is having some issues, do you want me to fix it on this branch, or do a different PR with this one needing a rebase afterwards?

**Comment by Arcanemagus** [comment]
That ought to do it, assuming OS X works properly.

**Comment by wooorm** [comment]
Looks like it? Wanna publish?

**Comment by Arcanemagus** [comment]
Sure, I can push out a new version in a few. Looks like this should be a major as the embedded version was a major?

**Comment by wooorm** [comment]
Yeah I‘d go with major!

**Comment by Arcanemagus** [comment]
Published as v3.0.0 🎉.

**Comment by wooorm** [comment]
Woohoo!

**Pull request merged**